plugins/api-docs/src/components/ApiExplorerTable/defaults.tsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { TableColumn, TableFilter } from '@backstage/core';
+import { EntityRefLink, EntityRefLinks } from '@backstage/plugin-catalog-react';
+import { ApiEntityV1alpha1, EntityName } from '@backstage/catalog-model';
+import { ApiTypeTitle } from '../ApiDefinitionCard';
+import { Chip } from '@material-ui/core';
+
+export type CustomizableTableProps = {
+  columns?: Array<any>;
+  filters?: Array<string | TableFilter>;
+};
+
+export type EntityRow = {
+  entity: ApiEntityV1alpha1;
+  resolved: {
+    name: string;
+    partOfSystemRelationTitle?: string;
+    partOfSystemRelations: EntityName[];
+    ownedByRelationsTitle?: string;
+    ownedByRelations: EntityName[];
+  };
+};
+
+export type NullableColumn = TableColumn<EntityRow> | undefined;
+
+export const defaultColumns: TableColumn<EntityRow>[] = [
+  {
```

Comment:
I'm wondering if using a map and keying off the column name could help leveraging typing better here? That might make it possible for something like `Record<ColumnName, Column>`. Then downstream both the ColumnName (union of strings) can be exposed through props for the table component etc.

Using a map instead of a@n array could also make it easier to merge custom fields maybe.

Reply:
Sorry for the late response, but thanks for taking a peek, @andrewthauer. I agree that using a map rather than a list here would make things more robust - I went ahead and made that change. 

I played around with the types, but I'm not sure that they're providing the kind of usefulness that you brought up. If I'm understanding correctly, we'd ideally want the type to tell us that we can specify our own custom fields in addition to the set of defaults that Backstage provides. Please excuse my naivete around types; I came up with [this](https://github.com/backstage/backstage/pull/4845/commits/09707b5be1ab810f1f5bc2598b667aa8827919e4#diff-4590ec23dc1220db8338e67442e35bcfcd5f0cea547db933acc0e6cb2ffd9994R37-R49).

Reply:
Sorry for the late response, but thanks for taking a peek, @andrewthauer. I agree that using a map for defining the defaults here would be more robust - I went ahead and made that change.

I played around with the types in an attempt to generate something useful like what you're proposing, but I feel it's not quite there yet. If I'm understanding correctly, we want the types to tell us that we can provide our own custom fields in addition to a set of defaults that Backstage provides. I came up with [this](https://github.com/backstage/backstage/pull/4845/commits/09707b5be1ab810f1f5bc2598b667aa8827919e4#diff-4590ec23dc1220db8338e67442e35bcfcd5f0cea547db933acc0e6cb2ffd9994R37-R49). I think that the fact that the default names are a subset of the more general key type (`string`) is potentially what's keeping it from being more useful.

Lastly, it would be great if we could make this reusable like you mentioned. We could move and expose whatever pattern we come up with here to somewhere more general. Perhaps the catalog frontend plugin is best suited for this?